lib/spack/llnl/util/filesystem.py

```diff
@@ -1819,3 +1820,26 @@ def remove_directory_contents(dir):
                 os.unlink(entry)
             else:
                 shutil.rmtree(entry)
+
+
+@contextmanager
+def keep_modification_time(*filenames):
+    """
+    Context manager to keep the modification timestamps of the input files.
+
+    Parameters:
+        *filenames: one or more files that must have their modification
+            timestamps unchanged
+    """
+    mtimes = {}
+    for f in filenames:
+        try:
+            mtimes[f] = os.path.getmtime(f)
+        except (IOError, OSError):
+            pass
+    yield
+    for f, mtime in mtimes.items():
+        try:
+            os.utime(f, (os.path.getatime(f), mtime))
+        except (IOError, OSError):
```

Comment:
I'm not sure what error there would be other than if the file were deleted. In that case I think an explicit test would be useful. Where possible I think it's best to let unexpected behavior halt the process.

Also since the test now checks for the case where the user deletes the file, I think it's worth mentioning that in the docstring of this method.